sqlmodel/main.py

```diff
@@ -408,14 +424,27 @@ def __init__(
 def get_sqlalchemy_type(field: FieldInfo) -> Any:
     type_: Optional[type] = field.annotation
 
-    # Resolve Optional fields
-    if type_ is not None and get_origin(type_) is Union:
+    # Resolve Optional/Union fields
+    if type_ is not None and get_origin(type_) in (types.UnionType, Union):
         bases = get_args(type_)
         if len(bases) > 2:
             raise RuntimeError(
                 "Cannot have a (non-optional) union as a SQL alchemy field"
             )
         type_ = bases[0]
+    # Resolve Annoted fields,
+    # like typing.Annotated[pydantic_core._pydantic_core.Url, 
+    # UrlConstraints(max_length=512, 
+    # allowed_schemes=['smb', 'ftp', 'file']) ]
+    if type_ is pydantic.AnyUrl:
+        meta =  field.metadata[0]
+        return AutoString(length=meta.max_length)
+    
+    if  get_origin(type_) is Annotated:
+        type2 = type_.__args__[0]
```

Comment:
Mypy does not like this line:

>  Item "type" of "Optional[type]" has no attribute "__args__"

Maybe use https://docs.python.org/3/library/typing.html#typing.get_args

Reply:
Thanks! [typing.get_args](https://docs.python.org/3/library/typing.html#typing.get_args) is much better.